packages/get-extension-wallet/src/types.ts

```diff
@@ -1,145 +1,31 @@
-}
-
-export interface IGetAlephiumWallet {
-  /**
-   * connect to a wallet
-   *
-   * when disconnected -
-   *  - if there is one pre-authorized wallet:
-   *      connect to it automatically
-   *  - if there are multiple pre-authorized wallets and a default wallet:
-   *      connect to it automatically
-   *  - if there are multiple pre-authorized wallets and no default wallet:
-   *      show them first in the list, ordered by last connect
-   *      (where "last connect" stands for the timestamp when the user
-   *      have chosen to connect to that wallet via the list)
-   *  - if there are no pre-authorized wallets:
-   *      show all wallets by dapp chosen order (default to random),
-   *      installed wallets first, then other available wallets below
-   *
-   * when connected -
-   *  - show the "connect to a wallet" list
-   *      1. display all available wallets respectively to the previously given options
-   *      2. allow users to disconnect the current wallet
-   *      3. modify default wallet settings
-   *
-   * @param options
-   * @returns Promise of `AlephiumWindowObject` if the user have chosen a
-   * wallet from the list, undefined when the list got closed without choosing
-   * a wallet (i.e. the user simply canceled the "connect to a wallet" request,
-   * or chose to install a wallet from the wallets-discovery list).
-   *
-   * Once connected to a wallet, clients can retrieve the wallet's
-   * `AlephiumWindowObject` instance by calling `getAlephium()`
-   */
-  connect(
-    options?: GetAlephiumWalletOptions,
-  ): Promise<AlephiumWindowObject | undefined>
-
-  /**
-   * disconnect from a wallet
-   * @param options
-   */
-  disconnect(options?: DisconnectOptions): Promise<boolean>
-
-  /**
-   * returns a list of browser-installed wallets (as opposed to browser-available wallets)
-   * @param options see `connect` `options` for `order`, `include` & `exclude`
-   */
-  getInstalledWallets(
-    options?: Omit<GetAlephiumWalletOptions, "showList" | "modalOptions">,
-  ): Promise<AlephiumWindowObject[]>
-
-  /**
-   * return last-chosen wallet `AlephiumWindowObject` instance,
-   * or default wrapper if disconnected
-   */
-  getAlephium(): AlephiumWindowObject
-}
-
-export type EventType = "addressesChanged" | "networkChanged"
-
-export type EventHandler = (data: any) => Promise<void> | void
-
-export type EnableOptions = EnableOptionsBase & {
-  showModal?: boolean
-}
+export type EnableOptions = EnableOptionsBase
 
 export abstract class AlephiumWindowObject extends InteractiveSignerProvider<EnableOptions> {
-  isPreauthorized: () => Promise<boolean>
-  on: (event: EventType, handleEvent: EventHandler) => void
-  off: (event: EventType, handleEvent: EventHandler) => void
+  abstract id: string
+  abstract name: string
+  abstract icon: string
+  abstract version: string
+
+  abstract isPreauthorized: (options: EnableOptions) => Promise<boolean>
+  enableIfPreauthorized = async (
```

Comment:
The function tries to restore existing connections upon events like page refresh, new tab for the same host etc. Any other naming suggestions?

Reply:
I think the name is self-explanatory. If someone doesn't know "preauthorize", maybe `enableIfConnected` also makes sense? WDYT?

Reply:
`enableIfConnected` sounds better for me 👍